### PR TITLE
create notebook and function for plot_pairwise_factor distributions

### PR DESCRIPTION
**Pull request by effieHAN**
Needed files are added. please review

### CODE REVIEW

File: skillmodels/visualize_factor_distributions.py [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+            each period (if state_ranges are not given explicitly) and to
+            estimate the distribution of the latent factors.
+        model_dict (dict): The model specification. See: :ref:'model_specs'
+        period (int): The selected period of the filtered states that are plotted.
+        combine_plots_in_grid (boolen): decide whether to retrun a one figure
+            containing subplots for each factor pair or a dictionary of
+            individual plots. Default True.
+        add_3d_plots (boolen):decide whether to add 3D plots in grid of plots
+            or in the dict of individual plots. Default False.
+        state_ranges (dict): The keys are the names of the latent factors.
```
Comment (hmgaudecker) [inline skillmodels/visualize_factor_distributions.py:40]: Outdated

File: skillmodels/visualize_factor_distributions.py [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+                    out[(fac1, fac2)] = grid[row][col].get_figure()
+
+    return out
+
+
+def _get_error_message(data, factors, plot_type):
+    summary = data[factors].describe().round(3).to_string()
+    tb = get_traceback()
+    msg = (
+        f"An error occured while trying to generate a {plot_type} for the "
```
Comment (hmgaudecker) [inline skillmodels/visualize_factor_distributions.py:151]:
```suggestion
        f"\n\n\nAn error occured while trying to generate a {plot_type} for the\n"
```

File: skillmodels/visualize_factor_distributions.py [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+        f"factors {factors}. Here is some information on the factors:\n{summary}\n\n"
```
Comment (hmgaudecker) [inline skillmodels/visualize_factor_distributions.py:152]:
```suggestion
        f"factors\n\n\n    {factors}\n\n\nHere is some information on the factors:\n\n\n{summary}\n\n\n"
```

File: skillmodels/visualize_factor_distributions.py (hunk 1 above)
Comment (janosg) [reply skillmodels/visualize_factor_distributions.py]: Done

**Review by janosg — APPROVED** [review]

**Pull request merged**